src/compiler/scanner.ts

```diff
@@ -2118,7 +2119,22 @@ namespace ts {
             return token;
         }
 
-        function getDirectiveFromComment(text: string) {
+        function appendIfCommentDirective(commentDirectives: CommentDirective[] | undefined, text: string, commentDirectiveRegEx: RegExp, lineStart = tokenPos) {

```

Comment:
Parameters on new line. Instead of making last parameter optional make it always needed and pass correct values in invocation.